### PR TITLE
Update TC Analysis template

### PR DESCRIPTION
**Pull request by forsyth2**
## Issue resolution
- Equivalent to https://github.com/E3SM-Project/e3sm_diags/pull/824

Select one: This pull request is...
- [ ] a bug fix: increment the patch version
- [x] a small improvement: increment the minor version
- [ ] an incompatible (non-backwards compatible) API change: increment the major version

## 1. Does this do what we want it to do?

Objectives:
- Make `tc_analysis.bash` more closely match the file in E3SM Diags.

Required:
- [x] Product Management: I have confirmed with the stakeholders that the objectives above are correct and complete.
- [x] Testing: I have added or modified at least one "min-case" configuration file to test this change. Every objective above is represented in at least one cfg.
- [x] Testing: I have considered likely and/or severe edge cases and have included them in testing.

If applicable:
- [ ] Testing: this pull request introduces an important feature or bug fix that we _must_ test often. I have updated the weekly-test configuration files, not just a "min-case" one.
- [ ] Testing: this pull request adds at least one new possible parameter to the cfg. I have tested using this parameter with and without any other parameter that may interact with it.

## 2. Are the implementation details accurate & efficient?

Required:
- [x] Logic: I have visually inspected the entire pull request myself.
- [x] Logic: I have left GitHub comments highlighting important pieces of code logic. I have had these code blocks reviewed by at least one other team member.

If applicable:
- [ ] Dependencies: This pull request introduces a new dependency. I have discussed this requirement with at least one other team member. The dependency is noted in `zppy/conda`, not just an `import` statement.

## 3. Is this well documented?

Required:
- [ ] Documentation: by looking at the docs, a new user could easily understand the functionality introduced by this pull request.

## 4. Is this code clean?

Required:
- [x] Readability: The code is as simple as possible and well-commented, such that a new team member could understand what's happening.
- [x] Pre-commit checks: All the pre-commits checks have passed.

If applicable:
- [ ] Software architecture: I have discussed relevant trade-offs in design decisions with at least one other team member. It is unlikely that this pull request will increase tech debt.

### CODE REVIEW

**Comment by forsyth2** [comment]
TODO:
- [ ] Remove `scratch` parameter elsewhere it appears. Will need to stay in `default.ini` for backwards compatibility.
- [ ] Test the `tc_analysis` min-case `cfg`s (currently using v2 data)
- [ ] Switch those min-case `cfg`s back to using v3 data and test again

**Comment by forsyth2** [comment]
@chengzhuzhang The relevant code changes are in `zppy/templates/tc_analysis.bash`. Could you please review those to make sure they match up with https://github.com/E3SM-Project/e3sm_diags/pull/824 as intended?

I tested all the min-case `cfg` files on v2 data and everything was fine. However, I ran into some issues on `v3`. `min_case_e3sm_diags_tc_analysis_parallel` and `min_case_e3sm_diags_tc_analysis_mvm_2` resulted in `ERROR (1)` for the `tc_analysis` job. `ERROR (1)` means the dat file is empty. `grep -in error <.o file>` doesn't return anything in either case. I'm not sure why this would only happen on some of the runs.

If you'd like to try this out yourself:

<details>

<summary> Testing directions </summary>

```
# Check out the branch
git fetch upstream
git checkout -b tc-analysis-updates upstream/tc-analysis-updates

# Set up e3sm diags dev environment
cd <e3sm_diags directory>
git checkout main
git fetch upstream
git reset --hard upstream/main
git log # Should match https://github.com/E3SM-Project/e3sm_diags/commits/main
mamba clean --all
mamba env create -f conda-env/dev.yml -n e3sm_diags_<date>
conda activate e3sm_diags_<date>
pip install .

# Set up zppy dev environment
cd <zppy directory>
mamba clean --all
mamba env create -f conda/dev.yml -n zppy_dev_pr632
conda activate zppy_dev_pr632
pip install .

# This block is only if you make further changes to the code
pre-commit run --all-files
pip install .
python -u -m unittest tests/test_*.py # Run unit tests

# Set up the testing files
emacs tests/integration/utils.py # set `UNIQUE_ID = "test-pr632"` and under `get_chrysalis_expansions`: `"diags_environment_commands": "source <path to conda.sh>; conda activate e3sm_diags_<date>",`
python tests/integration/utils.py # Actually generate the testing files

# Run zppy. These usually take 5-10 minutes to run in full.
zppy -c tests/integration/generated/test_min_case_e3sm_diags_tc_analysis_parallel_chrysalis.cfg
zppy -c tests/integration/generated/test_min_case_e3sm_diags_tc_analysis_mvm_1_chrysalis.cfg
# Wait for _1 to finish and then run:
zppy -c tests/integration/generated/test_min_case_e3sm_diags_tc_analysis_mvm_2_chrysalis.cfg

# Check results
cd /lcrc/group/e3sm/<username>/zppy_min_case_e3sm_diags_tc_analysis_parallel_output/test-pr632/v3.LR.historical_0051/post/scripts/
grep -v "OK" *status
cd /lcrc/group/e3sm/<username>/zppy_min_case_e3sm_diags_tc_analysis_mvm_2_output/test-pr632/v3.LR.historical_0051/post/scripts/
grep -v "OK" *status
```

</details>

**Comment by forsyth2** [comment]
> The relevant code changes are in `zppy/templates/tc_analysis.bash`. Could you please review those to make sure they match up with https://github.com/E3SM-Project/e3sm_diags/pull/824 as intended?

@chengzhuzhang Can you please review the changes in `zppy/templates/tc_analysis.bash`? If you approve those, I can merge this PR.

> I tested all the min-case `cfg` files on v2 data and everything was fine. However, I ran into some issues on `v3`. `min_case_e3sm_diags_tc_analysis_parallel` and `min_case_e3sm_diags_tc_analysis_mvm_2` resulted in `ERROR (1)` for the `tc_analysis` job. `ERROR (1)` means the dat file is empty. `grep -in error <.o file>` doesn't return anything in either case. I'm not sure why this would only happen on some of the runs.

I tested on v3 data, with this PR's code rebased on `main` (see details below). It looks like the change at https://github.com/E3SM-Project/zppy/pull/633/files#r1821703792 enables the code to move on past that error. The error is now the error E3SM Diags catches, but I believe that is what we want here, since we didn't change the input. So, I think this PR is ready for review.

<details>

<summary> Testing details </summary>

```
# Set up e3sm diags dev environment
cd ~/ez/e3sm_diags
git checkout main
git fetch upstream
git reset --hard upstream/main
git log
conda clean --all --y
conda env create -f conda-env/dev.yml -n e3sm_diags_main_20241210
conda activate e3sm_diags_main_20241210
pip install .

# Set up zppy dev environment
cd ~/ez/zppy
conda clean --all --y
conda env create -f conda/dev.yml -n zppy_dev_pr632_20241210
conda activate zppy_dev_pr632
pip install .

pytest tests/test_*.py # 23 passed, 1 warning in 0.40s

# Set up the testing files
emacs tests/integration/utils.py 
# set `UNIQUE_ID = "test-pr632-20241210"`
# under `get_chrysalis_expansions`: `"diags_environment_commands": "source /gpfs/fs1/home/ac.forsyth2/miniforge3/etc/profile.d/conda.sh; conda activate e3sm_diags_main_20241210",`

python tests/integration/utils.py # Actually generate the testing files
# CFG FILES HAVE BEEN GENERATED FROM TEMPLATES WITH THESE SETTINGS:
# UNIQUE_ID=test-pr632-20241210
# unified_testing=False
# diags_environment_commands=source /gpfs/fs1/home/ac.forsyth2/miniforge3/etc/profile.d/conda.sh; conda activate e3sm_diags_main_20241210
# global_time_series_environment_commands=source <INSERT PATH TO CONDA>/conda.sh; conda activate <INSERT ENV NAME>
# e3sm_to_cmip_environment_commands=
# environment_commands=


# tests/integration/template_min_case_e3sm_diags_tc_analysis.cfg
# tests/integration/template_min_case_e3sm_diags_tc_analysis_mvm_1.cfg
# tests/integration/template_min_case_e3sm_diags_tc_analysis_mvm_2.cfg
# tests/integration/template_min_case_e3sm_diags_tc_analysis_parallel.cfg

# tests/integration/template_min_case_e3sm_diags_tc_analysis_v2.cfg
# tests/integration/template_min_case_e3sm_diags_tc_analysis_v2_mvm_1.cfg
# tests/integration/template_min_case_e3sm_diags_tc_analysis_v2_mvm_2.cfg
# tests/integration/template_min_case_e3sm_diags_tc_analysis_v2_parallel.cfg


# Run zppy. These usually take 5-10 minutes to run in full.
zppy -c tests/integration/generated/test_min_case_e3sm_diags_tc_analysis_parallel_chrysalis.cfg
zppy -c tests/integration/generated/test_min_case_e3sm_diags_tc_analysis_mvm_1_chrysalis.cfg
# Wait for _1 to finish and then run:
zppy -c tests/integration/generated/test_min_case_e3sm_diags_tc_analysis_mvm_2_chrysalis.cfg

# Check results
cd /lcrc/group/e3sm/ac.forsyth2/zppy_min_case_e3sm_diags_tc_analysis_parallel_output/test-pr632-20241210/v3.LR.historical_0051/post/scripts/
grep -v "OK" *status
cd /lcrc/group/e3sm/ac.forsyth2/zppy_min_case_e3sm_diags_tc_analysis_mvm_2_output/test-pr632-20241210/v3.LR.historical_0051/post/scripts/
grep -v "OK" *status
```

</details>

<details>

<summary> Error details </summary>

No viewers were created:
- https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_min_case_e3sm_diags_tc_analysis_mvm_2_www/test-pr632-20241210/v3.LR.historical_0051/e3sm_diags/atm_monthly_180x360_aave_mvm/model_vs_model_1995-1996/ only has `prov/`; there's no Viewer generated
- https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_min_case_e3sm_diags_tc_analysis_parallel_www/test-pr632-20241210/v3.LR.historical_0051/e3sm_diags/atm_monthly_180x360_aave_tc_analysis/model_vs_obs_1985-1986/ also doesn't produce viewers
- https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_min_case_e3sm_diags_tc_analysis_parallel_www/test-pr632-20241210/v3.LR.historical_0051/e3sm_diags/atm_monthly_180x360_aave_tc_analysis/model_vs_obs_1987-1988/ also doesn't produce viewers

From `e3sm_diags_atm_monthly_180x360_aave_mvm_model_vs_model_1995-1996_vs_1985-1986.o643604`:
```
2024-12-10 12:29:36,739 [ERROR]: core_parameter.py(_run_diag:343) >> Error in e3sm_diags.driver.tc_analysis_driver
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.forsyth2/miniforge3/envs/e3sm_diags_main_20241210/lib/python3.12/site-packages/e3sm_diags/parameter/core_parameter.py", line 340, in _run_dia\
g
    single_result = module.run_diag(self)
                    ^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs/fs1/home/ac.forsyth2/miniforge3/envs/e3sm_diags_main_20241210/lib/python3.12/site-packages/e3sm_diags/driver/tc_analysis_driver.py", line 89, in run_diag
    test_data["metrics"] = generate_tc_metrics_from_te_stitch_file(test_te_file)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/gpfs/fs1/home/ac.forsyth2/miniforge3/envs/e3sm_diags_main_20241210/lib/python3.12/site-packages/e3sm_diags/driver/tc_analysis_driver.py", line 178, in generat\
e_tc_metrics_from_te_stitch_file
    raise ValueError(f"The file {te_stitch_file} is empty.")
ValueError: The file /lcrc/group/e3sm/ac.forsyth2/zppy_min_case_e3sm_diags_tc_analysis_mvm_2_output/test-pr632-20241210/v3.LR.historical_0051/post/atm/tc-analysis_1995\
_1996/cyclones_stitch_v3.LR.historical_0051_1995_1996.dat is empty.
2024-12-10 12:29:36,763 [WARNING]: e3sm_diags_driver.py(main:378) >> There was not a single valid diagnostics run, no viewer created.
2024-12-10 12:29:36,763 [ERROR]: run.py(run_diags:91) >> Error traceback:
Traceback (most recent call last):
  File "/gpfs/fs1/home/ac.forsyth2/miniforge3/envs/e3sm_diags_main_20241210/lib/python3.12/site-packages/e3sm_diags/run.py", line 89, in run_diags
    params_results = main(params)
                     ^^^^^^^^^^^^
  File "/gpfs/fs1/home/ac.forsyth2/miniforge3/envs/e3sm_diags_main_20241210/lib/python3.12/site-packages/e3sm_diags/e3sm_diags_driver.py", line 397, in main
    if parameters_results[0].fail_on_incomplete and (
       ~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```
We can see this IndexError is actually ultimately caused by the `ValueError` from https://github.com/E3SM-Project/e3sm_diags/commit/ffbdf287a91eca57fd383a1de4c0cf22e9fce829#diff-d256b358c8e18c01cd1de1ac39472fce1ba7c86f66a530ae321e5b8e33526eebR169.

</details>

File: zppy/templates/tc_analysis.bash [hunk 1]
```diff
@@ -29,19 +29,13 @@ start="{{ '%04d' % (year1) }}"
...
-# Warning: tempest-remap can only write grid file on SCRATCH space.
-# The result files will be moved to another path at the end.
-result_dir_fin={{ output }}/post/atm # Directory will be {{ output }}/post/atm/tc-analysis_${Y1}_${Y2}
-mkdir -p $result_dir_fin
 y1={{ year1 }}
 y2={{ year2 }}
 Y1="{{ '%04d' % (year1) }}"
 Y2="{{ '%04d' % (year2) }}"
-result_dir={{ scratch }}/tc-analysis_${Y1}_${Y2}/
+result_dir={{ output }}/post/atm/tc-analysis_${Y1}_${Y2}/
```
Comment (chengzhuzhang) [inline zppy/templates/tc_analysis.bash:36]: @forsyth2 the changes look good to me. With the fix in mesh generation, I think we no longer need extra step to save file in `scratch`? but it looks like all tc_analysis tests include `scratch`.

Comment (forsyth2) [reply zppy/templates/tc_analysis.bash:36]: Yes, that's correct. I just remove the `scratch` parameters from the test `cfg`s in the latest commit.
